{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "#Скачиваем собранный геном бактерии (все скаффолды)"
      ],
      "metadata": {
        "id": "xA-MRBx5dk-r"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://raw.githubusercontent.com/SanyaZmei133/hse22_hw1/main/data/scaffolds.fa"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f8wgEF6wcU1M",
        "outputId": "f2984b42-174c-4a6c-9fe7-713fb8548790"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 18:18:44--  https://raw.githubusercontent.com/SanyaZmei133/hse22_hw1/main/data/scaffolds.fa\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3924688 (3.7M) [text/plain]\n",
            "Saving to: ‘scaffolds.fa’\n",
            "\n",
            "scaffolds.fa        100%[===================>]   3.74M  --.-KB/s    in 0.03s   \n",
            "\n",
            "2022-10-19 18:18:44 (133 MB/s) - ‘scaffolds.fa’ saved [3924688/3924688]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Устанавливаем Genemask"
      ],
      "metadata": {
        "id": "Kf8UPcuqj0vW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_0APkf/gms2_linux_64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I3S9HRrycHhd",
        "outputId": "774e18e9-4e2f-47a4-f644-3c140a53b5ec"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 18:19:46--  http://topaz.gatech.edu/GeneMark/tmp/GMtool_0APkf/gms2_linux_64.tar.gz\n",
            "Resolving topaz.gatech.edu (topaz.gatech.edu)... 143.215.161.100\n",
            "Connecting to topaz.gatech.edu (topaz.gatech.edu)|143.215.161.100|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 11261753 (11M) [application/x-gzip]\n",
            "Saving to: ‘gms2_linux_64.tar.gz’\n",
            "\n",
            "gms2_linux_64.tar.g 100%[===================>]  10.74M  2.59MB/s    in 5.1s    \n",
            "\n",
            "2022-10-19 18:19:52 (2.10 MB/s) - ‘gms2_linux_64.tar.gz’ saved [11261753/11261753]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_0APkf/gm_key_64.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Mkw00VQ5cHPQ",
        "outputId": "55fae98a-8d70-471a-f656-0631789f31ce"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 18:20:15--  http://topaz.gatech.edu/GeneMark/tmp/GMtool_0APkf/gm_key_64.gz\n",
            "Resolving topaz.gatech.edu (topaz.gatech.edu)... 143.215.161.100\n",
            "Connecting to topaz.gatech.edu (topaz.gatech.edu)|143.215.161.100|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 85 [application/x-gzip]\n",
            "Saving to: ‘gm_key_64.gz’\n",
            "\n",
            "gm_key_64.gz        100%[===================>]      85  --.-KB/s    in 0s      \n",
            "\n",
            "2022-10-19 18:20:15 (12.2 MB/s) - ‘gm_key_64.gz’ saved [85/85]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d gm_key_64.gz"
      ],
      "metadata": {
        "id": "zzNFZ_3NkBN9"
      },
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!tar -xzvf gms2_linux_64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0lwJJ6d5kPR-",
        "outputId": "21dde824-5761-4da6-b380-cf64a76eada4"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gms2_linux_64/\n",
            "gms2_linux_64/gms2.pl\n",
            "gms2_linux_64/README_GMS2.txt\n",
            "gms2_linux_64/mgm_4.mod\n",
            "gms2_linux_64/gmhmmp2\n",
            "gms2_linux_64/INSTALL\n",
            "gms2_linux_64/mgm_11.mod\n",
            "gms2_linux_64/biogem\n",
            "gms2_linux_64/LICENSE\n",
            "gms2_linux_64/mgm_25.mod\n",
            "gms2_linux_64/compp\n",
            "gms2_linux_64/mgm_15.mod\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp  -v  gm_key_64   ~/.gmhmmp2_key"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ovT0SqWhkVUc",
        "outputId": "3f06fe2b-bb55-42b9-cc77-448a9fffaada"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'gm_key_64' -> '/root/.gmhmmp2_key'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Установка E-utilities (для скачивания последовательностей из NCBI)"
      ],
      "metadata": {
        "id": "PQghmLxgkf1n"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sh -c \"$(curl -fsSL ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LB45kKfjklAt",
        "outputId": "efef76e5-70dc-4357-f731-ba27871c6610"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Entrez Direct has been successfully downloaded and installed.\n",
            "\n",
            "In order to complete the configuration process, please execute the following:\n",
            "\n",
            "  echo \"export PATH=\\${PATH}:/root/edirect\" >> ${HOME}/.bashrc\n",
            "\n",
            "or manually edit the PATH variable assignment in your .bashrc file.\n",
            "\n",
            "Would you like to do that automatically now? [y/N]\n",
            "y\n",
            "OK, done.\n",
            "\n",
            "To activate EDirect for this terminal session, please execute the following:\n",
            "\n",
            "export PATH=${PATH}:${HOME}/edirect\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Скачиваем данные по близкородственной бактерии T.oleivorans"
      ],
      "metadata": {
        "id": "MXgdMGnEl2Tw"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Геном, последовательности генов (нт) и белков (протеом) для бактерии Thalassolituus oleivorans MIL-1 https://www.ncbi.nlm.nih.gov/nuccore/HF680312"
      ],
      "metadata": {
        "id": "jtCZ_JIvl3uQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format gb  >  T_oleivorans_MIL_1.gbk"
      ],
      "metadata": {
        "id": "rDQn1lzsl7RU"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format gene_fasta  >  T_oleivorans_MIL_1.genes.fasta"
      ],
      "metadata": {
        "id": "qkVMScbsmAt5"
      },
      "execution_count": 11,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format fasta_cds_aa  >  T_oleivorans_MIL_1.proteins.fasta"
      ],
      "metadata": {
        "id": "dHOlZGVrmBHJ"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Установка BLAST"
      ],
      "metadata": {
        "id": "ur1lB_j4mso5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jjJm9Txemt83",
        "outputId": "62f31ee1-4e44-42fd-b4d4-3eebb9f0e8f3"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]\n",
            "\r0% [Connecting to archive.ubuntu.com (91.189.91.38)] [Connecting to security.ub\r0% [Connecting to archive.ubuntu.com (91.189.91.38)] [Connecting to security.ub\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]\n",
            "\r0% [Connecting to archive.ubuntu.com (91.189.91.38)] [Connecting to security.ub\r0% [1 InRelease gpgv 15.9 kB] [Connecting to archive.ubuntu.com (91.189.91.38)]\r                                                                               \rHit:3 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n",
            "\r0% [1 InRelease gpgv 15.9 kB] [Connecting to archive.ubuntu.com (91.189.91.38)]\r                                                                               \rHit:4 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease\n",
            "\r0% [1 InRelease gpgv 15.9 kB] [Connecting to archive.ubuntu.com (91.189.91.38)]\r                                                                               \rHit:5 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n",
            "\r0% [1 InRelease gpgv 15.9 kB] [Connecting to archive.ubuntu.com (91.189.91.38)]\r                                                                               \rGet:6 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n",
            "Hit:7 http://archive.ubuntu.com/ubuntu bionic InRelease\n",
            "Get:8 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n",
            "Ign:9 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n",
            "Get:10 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main Sources [2,165 kB]\n",
            "Get:11 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease [1,581 B]\n",
            "Hit:12 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n",
            "Get:13 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main amd64 Packages [1,109 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [83.3 kB]\n",
            "Get:15 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ Packages [95.2 kB]\n",
            "Get:16 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [983 kB]\n",
            "Get:18 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [1,208 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,332 kB]\n",
            "Get:20 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [3,035 kB]\n",
            "Get:21 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,554 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [3,467 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [1,248 kB]\n",
            "Fetched 17.5 MB in 5s (3,797 kB/s)\n",
            "Reading package lists... Done\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install ncbi-blast+"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n5o5P56mmuzz",
        "outputId": "d18fa7bc-2a28-4dd6-894b-dac833bdb754"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  ncbi-data\n",
            "The following NEW packages will be installed:\n",
            "  ncbi-blast+ ncbi-data\n",
            "0 upgraded, 2 newly installed, 0 to remove and 37 not upgraded.\n",
            "Need to get 13.1 MB of archives.\n",
            "After this operation, 66.7 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ncbi-data all 6.1.20170106-2 [3,645 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/universe amd64 ncbi-blast+ amd64 2.6.0-1 [9,446 kB]\n",
            "Fetched 13.1 MB in 2s (8,047 kB/s)\n",
            "Selecting previously unselected package ncbi-data.\n",
            "(Reading database ... 123941 files and directories currently installed.)\n",
            "Preparing to unpack .../ncbi-data_6.1.20170106-2_all.deb ...\n",
            "Unpacking ncbi-data (6.1.20170106-2) ...\n",
            "Selecting previously unselected package ncbi-blast+.\n",
            "Preparing to unpack .../ncbi-blast+_2.6.0-1_amd64.deb ...\n",
            "Unpacking ncbi-blast+ (2.6.0-1) ...\n",
            "Setting up ncbi-data (6.1.20170106-2) ...\n",
            "Setting up ncbi-blast+ (2.6.0-1) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Скачиваем БД белков SwissProt"
      ],
      "metadata": {
        "id": "4XzBGEHIm4eq"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YUR3V-HEm8D7",
        "outputId": "7a34d5e2-9411-409b-b5b0-1901802e3447"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 18:22:18--  https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz\n",
            "Resolving ftp.uniprot.org (ftp.uniprot.org)... 128.175.240.195\n",
            "Connecting to ftp.uniprot.org (ftp.uniprot.org)|128.175.240.195|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 91303010 (87M) [application/x-gzip]\n",
            "Saving to: ‘uniprot_sprot.fasta.gz’\n",
            "\n",
            "uniprot_sprot.fasta 100%[===================>]  87.07M  27.7MB/s    in 3.1s    \n",
            "\n",
            "2022-10-19 18:22:22 (27.7 MB/s) - ‘uniprot_sprot.fasta.gz’ saved [91303010/91303010]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d uniprot_sprot.fasta.gz"
      ],
      "metadata": {
        "id": "T-IHUsQWm8bX"
      },
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Устанавливаем пакет seqkit"
      ],
      "metadata": {
        "id": "hwpvnrjpnbx6"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/shenwei356/seqkit/releases/download/v2.1.0/seqkit_linux_amd64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Dfe1ZLqYnc-9",
        "outputId": "2b3267a8-9998-46e4-a1b1-7f3a8f7be3af"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-19 18:22:35--  https://github.com/shenwei356/seqkit/releases/download/v2.1.0/seqkit_linux_amd64.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.121.3\n",
            "Connecting to github.com (github.com)|140.82.121.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/52715040/ea4018ec-70e8-40ac-a4aa-3326b04e07c4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221019%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221019T182235Z&X-Amz-Expires=300&X-Amz-Signature=ddb6b36d6ab4687d4ae7091832132f14d58d519ae14e0d01b1fd36af0cae92da&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=52715040&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2022-10-19 18:22:35--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/52715040/ea4018ec-70e8-40ac-a4aa-3326b04e07c4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221019%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221019T182235Z&X-Amz-Expires=300&X-Amz-Signature=ddb6b36d6ab4687d4ae7091832132f14d58d519ae14e0d01b1fd36af0cae92da&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=52715040&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 6650850 (6.3M) [application/octet-stream]\n",
            "Saving to: ‘seqkit_linux_amd64.tar.gz’\n",
            "\n",
            "seqkit_linux_amd64. 100%[===================>]   6.34M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2022-10-19 18:22:35 (62.4 MB/s) - ‘seqkit_linux_amd64.tar.gz’ saved [6650850/6650850]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!tar -xzvf seqkit_linux_amd64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z8suQZRzndkS",
        "outputId": "c2b89ac8-9c1f-4ee4-f5a7-e5d6486c12ae"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "seqkit\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!chmod a+x seqkit"
      ],
      "metadata": {
        "id": "dc7HOC5XnhD3"
      },
      "execution_count": 19,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "of0gG6NBoBRl"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Выполнение задания"
      ],
      "metadata": {
        "id": "0do2DukEoNLa"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "##Предсказать расположение белок-кодирующих генов в геноме нашей бактерии"
      ],
      "metadata": {
        "id": "b0Da0ClBoU5J"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!./gms2_linux_64/gms2.pl  --seq scaffolds.fa  --genome-type bacteria  --fnn genes.fasta  --faa proteins.fasta"
      ],
      "metadata": {
        "id": "bZAhNhTzoYek"
      },
      "execution_count": 21,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "##Определяем функции белков нашей бактерии через сравнение с белками из T.oleivorans MIL-1"
      ],
      "metadata": {
        "id": "DXuojxMaoyuP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!makeblastdb  -dbtype prot  -in T_oleivorans_MIL_1.proteins.fasta  -out T_oleivorans_MIL_1.proteins"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zs502J-Woz9Y",
        "outputId": "e3140d67-705c-4c53-b491-4f803c042812"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "\n",
            "Building a new DB, current time: 10/19/2022 18:25:39\n",
            "New DB name:   /content/T_oleivorans_MIL_1.proteins\n",
            "New DB title:  T_oleivorans_MIL_1.proteins.fasta\n",
            "Sequence type: Protein\n",
            "Keep MBits: T\n",
            "Maximum file size: 1000000000B\n",
            "Adding sequences from FASTA; added 3666 sequences in 0.104338 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!blastp  -query proteins.fasta  -db T_oleivorans_MIL_1.proteins  -evalue 1e-10  -outfmt 6  >  scaffolds.hits_from_MIL_1.txt"
      ],
      "metadata": {
        "id": "K-tk555no6-L"
      },
      "execution_count": 23,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "##Определяем функции белков нашей бактерии через поиск по БД SwissProt"
      ],
      "metadata": {
        "id": "UqS8AWbtghdr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cut -f 1 scaffolds.hits_from_MIL_1.txt | sort -n | uniq > proteins.with_hits_from_MIL_1.txt"
      ],
      "metadata": {
        "id": "nqySCrT-ggMa"
      },
      "execution_count": 24,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!./seqkit grep  --invert-match  -f proteins.with_hits_from_MIL_1.txt  proteins.fasta   -o proteins.without_MIL_1.fasta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3eyixm8jgu0N",
        "outputId": "3fd56828-555a-459b-bbea-6c26cd3c509b"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[INFO]\u001b[0m 3283 patterns loaded from file\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Общее количество белков бактерии"
      ],
      "metadata": {
        "id": "VscFeu6QhiaY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!grep '>' proteins.fasta | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CMfPGUX2g7S8",
        "outputId": "b9241f6d-bfe6-47b8-82f0-f524daf7899d"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "3566\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wc -l proteins.with_hits_from_MIL_1.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tsjFoN2BhAOr",
        "outputId": "99219c95-ce08-4bb4-ae57-89d4ba5013db"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "3283 proteins.with_hits_from_MIL_1.txt\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!grep '>' proteins.without_MIL_1.fasta | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gAO95Olche2k",
        "outputId": "92616756-69aa-4bf7-d18d-096c2e93db38"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "283\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!makeblastdb  -dbtype prot  -in uniprot_sprot.fasta  -out uniprot_sprot"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pRPYvSCXhyoD",
        "outputId": "5cef6e51-d942-4e20-861c-777fd7db240e"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "\n",
            "Building a new DB, current time: 10/19/2022 18:35:14\n",
            "New DB name:   /content/uniprot_sprot\n",
            "New DB title:  uniprot_sprot.fasta\n",
            "Sequence type: Protein\n",
            "Keep MBits: T\n",
            "Maximum file size: 1000000000B\n",
            "Adding sequences from FASTA; added 568363 sequences in 12.9617 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Определеям какие белки из нашей бактерии похожи на белки из БД SwissProt"
      ],
      "metadata": {
        "id": "rs-oU0VTh7PF"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!time  blastp  -query proteins.without_MIL_1.fasta  -db uniprot_sprot  -evalue 1e-10  -outfmt 6  >  scaffolds.hits_from_SwissProt.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9KYWfSzDh5bb",
        "outputId": "c57aa326-ad40-45ff-b43b-528389214b1c"
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "real\t9m27.047s\n",
            "user\t9m25.161s\n",
            "sys\t0m0.410s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Посмотрим, сколько белков удалось аннотировать через поиск по SwissProt:"
      ],
      "metadata": {
        "id": "PvHfOCdrkQ9V"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cut -f 1 scaffolds.hits_from_SwissProt.txt | sort | uniq | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5y1d8b91kLpe",
        "outputId": "3bfc19ed-d09f-4272-8fea-eca3d8ac6103"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "53\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##Загружаем файлы, которые нам потребуются для аннотации генома, на github"
      ],
      "metadata": {
        "id": "yrmEQuh2llrh"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "* gms2.lst -- координаты всех предсказанных генов\n",
        "* proteins.fasta -- аминокислотные последовательности всех предсказанных генов\n",
        "* scaffolds.hits_from_MIL_1.txt -- информация о схожести белков из нашей бактерии с белками из бактерии MIL-1\n",
        "* scaffolds.hits_from_SwissProt.txt -- информация о схожести белков из нашей бактерии с белками из БД SwissProt"
      ],
      "metadata": {
        "id": "4MMdmrdDg2pc"
      }
    }
  ]
}